Nimble/CallRecorder.swift

```diff
@@ -0,0 +1,312 @@
+
+/* bug in Swift causes every enum WITHOUT an associated value's "description" to be the first declared
+enum value WITHOUT an associated value's description
+i.e. -> since ".DontCare" is the first enum value then the default description for ".DontCare",
+".NonNil", and ".Nil" will all be "DontCare" -> must override to fix issue by conforming to "CustomStringConvertible" 
+    This is not reproducable in a Playground */
+
+public enum Argument : CustomStringConvertible {
+    case DontCare
+    case NonNil
+    case Nil
+    case InstanceOf(type: Any.Type)
+    case InstanceOfWith(type: Any.Type, option: ArgumentOption)
+    case KindOf(type: AnyObject.Type)
+    
+    public var description: String {
+        switch self {
+        case .DontCare:
+            return "Argument.DontCare"
+        case .NonNil:
+            return "Argument.NonNil"
+        case .Nil:
+            return "Argument.Nil"
+        case .InstanceOf(let type):
+            return "Argument.InstanceOf(\(type))"
+        case .InstanceOfWith(let input):
+            return "Argument.InstanceOfWith(\(input.type), \(input.option))"
+        case .KindOf(let type):
+            return "Argument.KindOf(\(type))"
+        }
+    }
+}
+
+public enum ArgumentOption : CustomStringConvertible {
+    case DontCare
+    case NonOptional
+    case Optional
+    
+    public var description: String {
+        switch self {
+        case .DontCare:
+            return "ArgumentOption.DontCare"
+        case .NonOptional:
+            return "ArgumentOption.NonOptional"
+        case .Optional:
+            return "ArgumentOption.Optional"
+        }
+    }
+}
+
+public struct DidCallResult {
+    public let success: Bool
+    public let recordedCallsDescription: String
+//    public let successfullyCalledCount: Int
+}
+
+public enum DidCallResultIncludeOption : CustomStringConvertible {
+    case Yes
+    case No
+    case OnlyOnSuccess
+    case OnlyOnUnsuccess
+    
+    public var description: String {
+        switch self {
+        case .Yes:
+            return "DidCallResultIncludeOption.Yes"
+        case .No:
+            return "DidCallResultIncludeOption.No"
+        case .OnlyOnSuccess:
+            return "DidCallResultIncludeOption.OnlyOnSuccess"
+        case .OnlyOnUnsuccess:
+            return "DidCallResultIncludeOption.OnlyOnUnsuccess"
+        }
+    }
+}
+
+public protocol CallRecorder : class {
+    // For Interal Use ONLY -> Implement as empty properties when conforming to protocol
+    // Implementation Example:
+    // var called = (functionList: [String](), argumentsList: [[Any]]())
+    var called: (functionList: [String], argumentsList: [[Any]]) {get set}
+    
+    // **MUST** call in every method you want to spy
+    func recordCall(function function: String, arguments: Any...)
+    
+    // Used if you want to reset the called function/arguments lists
+    func clearRecordedLists()
+    
+    
+    // For Internal Use ONLY
```

Comment:
To be honest I'm not sure there is a better way. The mocked object needs to conform to this so that the matcher can call these functions on the mocked object. If you have an idea I'm willing to explore it, but as of now I'm not sure what I could do. That being said I'll think on it and see what/if I can come up with anything.